boot.py

```diff
@@ -207,19 +209,6 @@ def plugin_unloaded() -> None:
 
 
 class Listener(sublime_plugin.EventListener):
-    def _register_windows(self) -> None:
-        for w in sublime.windows():
-            windows.lookup(w)
-
-    def __del__(self) -> None:
```

Comment:
when does an EventListener get destroyed?

Reply:
It is destroyed when the plugin is loaded, not when it is unloaded. It was my assumption that these global listener objects are destroyed when the plugin unloads, but that is not the case. You can test this by creating a folder Packages/Foo with a file foo.py with the content

```python
import sublime_plugin


class Listener(sublime_plugin.EventListener):

    def __init__(self):
        print("__init__")

    def __del__(self):
        print("__del__")
```
If you disable the "Foo" package with package control, you won't see `__del__` printed to the console. Only when you enable the package again do you see `__del__` printed, followed immediately by `__init__`.